ena_runner.py

```diff
@@ -100,40 +100,22 @@ def create_batch(
 
             if metadata:
                 # Paired fastq for Illumina, single fastq for nanopore
-                if sample_method:
-                    if (
-                        sample_method == "illumina"
-                        and len(set((new_dir_prefix / dir).glob("*.fastq.md5"))) != 2
-                    ):
-                        print(
-                            f"{dir} is not a valid illumina sample, only one on the fastqs is avalaible."
-                        )
-                        validSample = False
-                    elif (
-                        sample_method == "nanopore"
-                        and len(set((new_dir_prefix / dir).glob("*.fastq.md5"))) != 1
-                    ):
-                        print(
-                            f"{dir} is not a valid nanopore sample, there is more than one fastq."
-                        )
-                        validSample = False
-                elif "instrument_platform" in metadata:
-                    if (
-                        str(metadata["instrument_platform"]).lower() == "illumina"
-                        and len(set((new_dir_prefix / dir).glob("*.fastq.gz"))) != 2
-                    ):
-                        print(
-                            f"{dir} is not a valid illumina sample, only one on the fastqs is avalaible."
-                        )
-                        validSample = False
-                    elif (
-                        sample_method == "nanopore"
-                        and len(set((new_dir_prefix / dir).glob("*.fastq.gz"))) != 1
-                    ):
-                        print(
-                            f"{dir} is not a valid nanopore sample, there is more than one fastq."
-                        )
-                        validSample = False
+                if sample_method or "instrument_platform" in metadata:
+                    
+                    # ILLUMINA
+                    if (sample_method == "illumina") or str(metadata["instrument_platform"]).lower() == "illumina":
+                        # check both fastqs are avalaible
+                        if len(set((new_dir_prefix / dir).glob("*.fastq.md5"))) != 2:
+                            print(f"{dir} is not a valid illumina sample, only one on the fastqs is avalaible.")
+                            validSample = False
+
+                    # NANOPORE
+                    elif (sample_method == "nanopore") or str(metadata["instrument_platform"]).lower() == "nanopore":
+                        # check that only one fastq exists
+                        if len(set((new_dir_prefix / dir).glob("*.fastq.md5"))) != 1:
```

Comment:
Are you sure you want to use _set_? set is the devil
in case you have to search for files in different folders, the files could have the same name, but the _set_ method would say that there is only one file. I would use _list_ instead. 
If the described scenario is too far fetched, close this comment

Reply:
Fien as is, should not happen